src/pages/home/home.ts

```diff
@@ -1,45 +1,132 @@
+  GoogleMap,
+  GoogleMapsEvent,
+  GoogleMapOptions,
+  CameraPosition,
+  MarkerOptions,
+  Marker,
+} from '@ionic-native/google-maps';
+import { Platform } from 'ionic-angular';
 import { YelpService } from '../../app/services/yelp/yelp.service';
 import { Observable } from 'rxjs/Observable';
 import { Geolocation } from '@ionic-native/geolocation';
 import { WorkfromService } from '../../app/services/workfrom/workfrom.service';
 
 @Component({
   selector: 'page-home',
-  templateUrl: 'home.html'
+  templateUrl: 'home.html',
 })
 export class HomePage {
-
   yelpLocations$: Observable<any>;
   workfromLocations$: Observable<any>;
 
   latitude: number;
   longitude: number;
 
+  map: GoogleMap;
+  mapElement: HTMLElement;
+
   constructor(
-    navCtrl: NavController,
-    platform: Platform,
+    private googleMaps: GoogleMaps,
+    public platform: Platform,
     private yelpService: YelpService,
     private workfromService: WorkfromService,
     private geolocation: Geolocation
-  ) {
-    platform.ready()
-      .then(() => this.getCurrentPosition());
+  ) { }
+
+  ngAfterViewInit() {
+    console.log('Ion view loaded.');
+    this.platform
+      .ready()
+      .then(() => this.getCurrentPosition())
+      .then((position) => this.loadMap(position))
+      .then((position) => this.getWorkfromLocations(position));
   }
 
-  getCurrentPosition() {
-    this.geolocation.getCurrentPosition()
-    .then(resp => {
-      this.latitude = resp.coords.latitude;
-      this.longitude = resp.coords.longitude;
 
-      this.yelpLocations$ = this.yelpService.getBusinesses(this.latitude, this.longitude);
-      this.workfromLocations$ = this.workfromService.getPlaces(this.latitude, this.longitude);
+  loadMap(position) {
+    this.mapElement = document.getElementById('map');
+    const currentLat = position.coords.latitude;
+    const currentLng = position.coords.longitude;
+    alert(currentLat)
+    alert(currentLng)
 
-    }).catch(error => {
-      console.log('Error getting geolocation', error);
+    let mapOptions: GoogleMapOptions = {
+      camera: {
+        target: {
+          lat: currentLat,
+          lng: currentLng,
+        },
+        zoom: 18,
+        tilt: 30,
+      },
+    };
+
+    this.map = this.googleMaps.create(this.mapElement, mapOptions);
+
+    // Wait the MAP_READY before using any methods.
+    this.map.one(GoogleMapsEvent.MAP_READY).then(() => {
+      console.log('Map is ready!');
+
+      // Now you can use all methods safely.
+      this.map
+        .addMarker({
+          title: 'Ionic',
+          icon: 'blue',
+          animation: 'DROP',
+          position: {
+            lat: currentLat,
+            lng: currentLng,
+          },
+        })
+        .then(marker => {
+          marker.on(GoogleMapsEvent.MARKER_CLICK).subscribe(() => {
+            alert('clicked');
+          });
+        });
     });
+
+    return position;
   }
 
+  getCurrentPosition() {
+    if (navigator.geolocation) {
```

Comment:
I removed the if statement. I think that if an error occurs, it will be propagated onto the catch I added in NgAfterViewInit. Let me know if you think this adequately resolves your concern.